CMake/AddDependencies.cmake

```diff
@@ -0,0 +1,85 @@
+function(build_dependencies TARGET)
```

Comment:
I don't think the `add_subdirectories` call makes sense as a function of the target. You couldn't call it for multiple targets for example. Maybe split this into two functions.